Add fuzzing function for Parse()

fernape opened the pull request:
This function will receive inputs with a high variability. The fuzzer should help us catch any problems early as we complete the parser.

Run with `go test --fuzz=Fuzz`

benhoyt commented on internal/parse/lexer_test.go:
```diff
@@ -83,6 +83,29 @@ FROM person`
+
+	assert.Equal(t, STRING, tokens[1].Type)
+	assert.Equal(t, "'s", tokens[1].Literal)
+	assert.Equal(t, 8, tokens[1].Pos.Column)
```
Should we not test for an error at the end as well? Presumably unterminated string is an error?

fernape replied:
The lexer does not return errors. [Everything](https://github.com/canonical/sqlair/blob/1a38304ffea770462f250f3e655132e3e58dd0b6/internal/parse/lexer.go#L129) is handled [downstream](https://github.com/canonical/sqlair/blob/1a38304ffea770462f250f3e655132e3e58dd0b6/internal/parse/lexer.go#L166)

benhoyt commented on internal/parse/lexer_test.go:
```diff
@@ -83,6 +83,29 @@ FROM person`
+
+	tokens := tokensForStatement(stmt)
+
+	assert.Equal(t, STRING, tokens[1].Type)
```
I'm curious why the arguments are reversed (here and in other places)? Normally you'd say `assert.Equal(t, tokens[1].Type, STRING)` so it reads more like the English order: "assert that tokens[1]'s Type field equals STRING".

fernape replied:
The second parameter is the expected value and the third one the actual value.
E.g:

```
assert.Equal(t, 1, 2)
```

```
--- FAIL: TestLexerUnterminatedString (0.00s)
    lexer_test.go:94: 
                Error Trace:    lexer_test.go:94
                Error:          Not equal: 
                                expected: 1
                                actual  : 2
                Test:           TestLexerUnterminatedString
```

benhoyt replied:
Oh, interesting, I hadn't realized that -- thanks!

manadart reviewed (approved):
Nice; thanks.

fernape committed:
Add fuzzing function for Parse()

This function will receive inputs with a high variability. The fuzzer
should help us catch any problems early as we complete the parser.

Run with go test --fuzz=Fuzz

niemeyer reviewed (approved):
For other reviewers, using "Hide whitespace" will show that most of the PR is just moving the tests around without changing them.

LGTM, thanks for the fuzzing.

manadart commented:
@Aflynn50 Prefer rebasing in-flight branches over merging the mainline back in. It makes for a cleaner commit history.

Aflynn50 committed:
Unbreak last update.

niemeyer commented:
Thanks for fixing it, Alastair.
